boards/PDM/Src/Io/Io_Efuse.c

```diff
@@ -0,0 +1,201 @@
+#include "Io_Efuse.h"
+
+#define EFUSE_ADDR_MASK 0xFU
+#define EFUSE_ADDR_SHIFT 0x0AU
+#define EFUSE_SI_DATA_MASK 0x1FFU
+#define EFUSE_SO_DATA_MASK 0xFF // Ignore Normal-Mode status bit (bit 9)
+
+#define WATCH_DOG_BIT (1U << 15U)
+#define PARITY_BIT (1U << 14U)
+
+// The SPI handle for the SPI device the E-Fuses are connected to
+static SPI_HandleTypeDef *efuse_spi_handle;
+
+void Io_Efuse_Init(SPI_HandleTypeDef *const hspi)
+{
+    efuse_spi_handle = hspi;
+}
+
+void Io_Efuse_ConfigureEfuse(struct Efuse *e_fuse)
```

Comment:
I feel like this should be rewritten with arguments that should be initialized, rather than relying on the user knowing which arguments should and should not be initialized. Adding some example arguments but I think you get the drift.
```suggestion
void Io_Efuse_ConfigureEfuse(struct Efuse **e_fuse, GPIO_Port chip_select_port, GPIO_Pin chip_select_pin...)
```
This way we can place `struct Efuse` in the `.c` and hide away details the user doesn't need to know

Reply:
Currently in the app layer you create the efuse like this
```
   e_fuse1 = App_E_Fuse_Create(
        CSB_AUX1_AUX2_GPIO_Port, CSB_AUX1_AUX2_Pin,
        CUR_SYNC_AUX1_AUX2_GPIO_Port, CUR_SYNC_AUX1_AUX2_Pin);
    e_fuse2 = App_E_Fuse_Create(
        CSB_EM_PUMP_GPIO_Port, CSB_EM_PUMP_Pin, CUR_SYNC_EM_PUMP_GPIO_Port,
        CUR_SYNC_EM_PUMP_Pin);
    e_fuse3 = App_E_Fuse_Create(
        CSB_DI_L_DI_R_GPIO_Port, CSB_DI_L_DI_R_Pin,
        CUR_SYNC_DI_L_DI_R_GPIO_Port, CUR_SYNC_DI_L_DI_R_Pin);
    e_fuse4 = App_E_Fuse_Create(
        CSB_GLV_AIR_SHDN_GPIO_Port, CSB_GLV_AIR_SHDN_Pin,
        CUR_SYNC_GLV_AIR_SHDN_GPIO_Port, CUR_SYNC_GLV_AIR_SHDN_Pin);
    e_fuses = App_E_Fuses_Create(e_fuse1, e_fuse2, e_fuse3, e_fuse4);